### PR TITLE
Bug recreation with suggested fixes for #209

### PR DESCRIPTION
**Pull request by WillGunn**
Recreated the bug we multiple parameters colliding and overriding the request body example provided by SwaggerRequestExample.  In this case because we had a header that was of type string and one of the examples in the example repository was of type string and ServiceProviderExamplesOperationFilter is setup to run after ExamplesOperationFilter in SwaggerGenOptionsExtensions.ExampleFilters; the last parameter in the parameterDescription list with a matching IExamplesProvider clobbers any existing example that was set by ExamplesOperationFilter even if it is not of type BindingSource.Body.

Suggested fix is to check the Source parameter on the parameterDescription and only Set the RequestBody example if it matches.  It may also be prudent to make ExamplesOperationFilter run after ServiceProviderExamplesOperationFilter, that way the last example value that gets set is the one manually specified via the attributes, but I'm not sure if you specifically intended for ServiceProviderExamplesOperationFilter to run after ExamplesOperationFilter or not.

There may be a related bug in the SetResponseExamples function, but I haven't explored that.

See #209

### CODE REVIEW

File: test/WebApi2.1-Swashbuckle5/Startup.cs [hunk 1]
```diff
@@ -24,19 +29,31 @@ public Startup(IConfiguration configuration)
         public void ConfigureServices(IServiceCollection services)
         {
             services
-                .AddMvc()
+                .AddMvcCore()
+                .AddApiExplorer()
+                .AddVersionedApiExplorer(options => options.GroupNameFormat = "'v'VVV")
+                .AddAuthorization()
```
Comment (mattfrear) [inline test/WebApi2.1-Swashbuckle5/Startup.cs:35]: Can you please not call so many .Add() methods

**Comment by mattfrear** [comment]
Hi @WillGunn . 

I think of the 2 fixes I prefer fix #1, as swapping the order of filters might cause other issues.
Can you please fix the unit tests too.

Thanks

**Comment by WillGunn** [comment]
@mattfrear so would you like me to keep the reproduction changes to the test app?  Or remove those and just implement fix #1?

File: test/WebApi2.1-Swashbuckle5/Startup.cs (hunk 1 above)
Comment (WillGunn) [reply test/WebApi2.1-Swashbuckle5/Startup.cs:35]: Previously you had the group call AddMvc which internally essentially calls what I put there, but with the ability to customize the version format.  I'll see if its still reproduceable with the group call AddMvc

**Comment by mattfrear** [comment]
Hi @WillGunn. I've just stepped through your code to understand the problem better.

Good question about the BindingSource, I'm not sure, let's go with Body, Form and Custom for now.

- Please leave your repro in the test app, it's handy. 
- As you can see from the appveyor CI build, the solution doesn't yet compile.
- There are also a bunch of failing unit tests in the `Swashbuckle.AspNetCore.Filters.Test` solution. If you're not familiar with tests you can run them with Visual Studio Test Explorer.

**Comment by WillGunn** [comment]
while experimenting with the suggested fix in our main repo, we were still running into issues due to the explicit ExamplesOperationFilter being clobbered by examples found by ServiceProviderExamplesOperationFilter.  So I included a 2nd swaggeroptions extension method to swap the order, giving priority to the explicit ExamplesOperationFilter.  This leaves the original order as the original method for potential backwards compatibility issues.

**Comment by mattfrear** [comment]
Hi @WillGunn sorry I've been slow picking this up, I don't have a lot of spare time.
I downloaded your branch and then merged master into it. There were some compilation errors on the WebApi3.1-Swashbuckle5 test project so I have fixed those.

The WebApi3.1 project also has CSV support which was added recently via another pull request.

Unfortunately your branch crashes when I try to run the WebApi3.1 test project - it seems to be caused by your StringResponseExample not playing nicely with the CSV support.

I have pushed my findings so far into another branch in my repo.

I don't know how to properly collaborate in GitHub... I guess can you pull my branch into your fork? Or otherwise, can you make me a contributor in your fork?

**Comment by mattfrear** [comment]
> Unfortunately your branch crashes when I try to run the WebApi3.1 test project - it seems to be caused by your StringResponseExample not playing nicely with the CSV support.

I have added a workaround for this and pushed it to my branch.

I will continue to review this PR when I get spare time.

**Comment by mattfrear** [comment]
Released https://www.nuget.org/packages/Swashbuckle.AspNetCore.Filters/7.0.5

**Pull request merged**